core/src/main/java/org/mule/runtime/core/context/notification/TransactionNotification.java

```diff
@@ -6,11 +6,15 @@
  */
 package org.mule.runtime.core.context.notification;
 
+import org.mule.runtime.core.api.context.notification.EnrichedNotificationInfo;
+import org.mule.runtime.core.api.context.notification.EnrichedServerNotification;
 import org.mule.runtime.core.api.context.notification.SynchronousServerEvent;
-import org.mule.runtime.core.api.context.notification.ServerNotification;
 import org.mule.runtime.core.api.transaction.Transaction;
 
-public class TransactionNotification extends ServerNotification implements SynchronousServerEvent {
+import java.util.HashMap;
+
+// Shouldn't really be an EnrichedServerNotification as it doesn't know event data, but inherits to be compatible with others
+public class TransactionNotification extends EnrichedServerNotification implements SynchronousServerEvent {
```

Comment:
Why can't you have another abstract class without the event data and extend from that one?